Update Get Apps card to reflect the Jetpack app

SiobhyB opened the pull request:
#### Changes proposed in this Pull Request:

The following changes have been made to the "get apps" section found via _Jetpack_ → _Dashboard_:

* The section is no longer dismissable.
* A QR code is available for users to scan when viewed on desktop, with buttons on mobile.
* Some updates have been made to the design, including tweaks to padding, the new app icon, etc.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

pcdRpT-F0-p2 

#### Does this pull request change what data or activity we track or use?

Yes, downloads from the app buttons may decrease, due to the fact that they'll only available from mobile devices following this change.

#### Testing instructions:

* With this branch checked out, navigate to _Jetpack_ → _Dashboard_.
* Scroll down to the mobile app card and verify that the design has been updated.

#### Screenshots:

<details>
 <summary>Design comparison (web) ⤵️</summary>

Note, the design in Figma used the `SF Pro` font. I didn't specify that in the design to keep consistency with the default font used in other sections on the page, but am happy to change if required. I aimed to match all other sizes/dimensions provided in the Figma.

**Design:**

<img width="879" alt="Screenshot 2022-09-20 at 16 49 20" src="https://user-images.githubusercontent.com/2998162/191305112-0239036b-e743-49ee-adcf-1801713aca0a.png">

**Actual:**

<img width="1023" alt="Screenshot 2022-09-20 at 17 07 18" src="https://user-images.githubusercontent.com/2998162/192515088-34954ff5-4744-4d8d-b530-4061569c33ff.png">

</details>

<details>
 <summary>Before/after (web) ⤵️</summary>

These before/after screenshots are taken in Google Chrome `(105.0.5195.125)` on macOS `(12.6)`.

**Before:**

<img width="1011" alt="Screenshot 2022-09-19 at 13 40 50" src="https://user-images.githubusercontent.com/2998162/191019602-3fa00c62-72db-4436-9936-083c380bdca8.png">

**After:**

<img width="1023" alt="Screenshot 2022-09-20 at 17 07 18" src="https://user-images.githubusercontent.com/2998162/192515088-34954ff5-4744-4d8d-b530-4061569c33ff.png">

</details>

<details>

 <summary>Mobile web ⤵️</summary>

It's worth noting that these screenshots were taken in emulators, rather than from actual devices.

| Google Pixel 5  | iPhone XR |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/192515528-eb6e87a9-9514-4995-af6a-da73e9ac3bbd.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/192515627-5ca3415b-3ec9-4490-a740-e2b0c1a67bbd.png" width="100%"> |
</details>

<details>
 <summary>x-browser testing ⤵️</summary>

The main browser I used for development was Chrome, for which screenshots have already been provided above. In addition, I tested this change across the latest versions of Opera, Firefox, Edge, and Safari on macOS `(12.6)`.

| Opera  | Firefox |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/192515901-f02009ec-15b3-4196-a90c-6135b98cd006.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/192516343-3afe7bf1-2691-40d6-99cf-e262ea1162e4.png" width="100%"> |

| Safari  | Edge |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/192516470-ea3ebcc9-cc9a-496f-a83e-9077772c57d3.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/192516690-08471929-e6bd-4962-ac66-9b99a37ff36e.png" width="100%"> |

</details>

SiobhyB committed:
Replace illustration in heading with app icon

SiobhyB committed:
Add changelog entry for changes to Get Apps section

SiobhyB committed:
Tweak styling to match designs

SiobhyB committed:
Add QR code, including copy for usage

SiobhyB committed:
Fix variable references

SiobhyB committed:
Import QRCode component for use

SiobhyB committed:
Fix incorrect variable reference

SiobhyB committed:
Fix incorrect variable reference

SiobhyB committed:
Update banner's icon with up-to-date design

SiobhyB committed:
Refactor to display cards on mobile, QR on web

SiobhyB committed:
Import 'isMobile' conditional

SiobhyB committed:
Remove option to dismiss card

SiobhyB committed:
Apply similar styles to both QR code and apps

shaunandrews commented:
I noticed the QR code is blurry:

<img width="210" alt="CleanShot 2022-09-26 at 15 47 21@2x" src="https://user-images.githubusercontent.com/191598/192366504-b0a7b410-90a4-408d-872e-9182d476d70c.png">

Maybe this is an artifact of you screenshots (BTW, thank you for the excellent screenshots!) or perhaps the QR code wasn't exported properly? Actually, it looks like a different QR from the one in the Figma; Is that intentional?

Here's an SVG if it helps:

![jp-com-app](https://user-images.githubusercontent.com/191598/192366933-5640b639-340b-48f7-a555-c60fac046c6c.svg)

SiobhyB committed:
Replace QRCode component with plan image HTML

This change has been made in response to the following feedback, to avoid blurriness: https://github.com/Automattic/jetpack/pull/26276#issuecomment-1258541243

SiobhyB committed:
Tweaks to font weight and sizes

SiobhyB commented:
@shaunandrews, thank you for checking! The code was blurry as I was using a custom component to generate it, rather than your `.svg`. I've gone ahead to update that now, and updated the screenshots also. Let me know how that looks to you and if any further changes are needed.

SiobhyB committed:
Merge branch 'trunk' into rnmobile/update-get-apps-section

dcalhoun commented on projects/plugins/jetpack/_inc/client/components/apps-card/index.jsx:
```diff
@@ -24,17 +23,40 @@ class AppsCard extends React.Component {
+				altText={ __( 'Apple App Store download badge.', 'jetpack' ) }
+				titleText={ __( 'Download the Jetpack iOS mobile app.', 'jetpack' ) }
+				storeName="ios"
+				storeLink="https://apps.apple.com/us/app/jetpack-wp-security-speed/id1565481562?pt=299112ct=jpdash&mt=8"
```
While working on jetpack.com/mobile, it was requested we leverage this specific URL for SEO purposes. 
```suggestion
				storeLink="https://apps.apple.com/us/app/jetpack-website-builder/id1565481562?pt=299112ct=jpdash&mt=8"
```

dcalhoun commented on projects/plugins/jetpack/_inc/client/components/apps-card/index.jsx:
```diff
@@ -24,17 +23,40 @@ class AppsCard extends React.Component {
+				<a className="jp-apps-card__link" href="https://jetpack.com/app">
+					jetpack.com/app
+				</a>
+				{ __( ' or scan this code to download the Jetpack mobile app.', 'jetpack' ) }
```
Explicitly directing users to use their mobile device may help avoid confusion that the URL does not work for non-mobile devices. pcdRpT-SK-p2#comment-1112 
```suggestion
				{ __( ' from your mobile device or scan this code to download the Jetpack mobile app.', 'jetpack' ) }
```

dcalhoun commented on projects/plugins/jetpack/_inc/client/components/apps-card/style.scss:
```diff
@@ -26,22 +26,21 @@
 
 .jp-apps-card__description {
 	max-width: 80%;
 	margin: 0 auto;
-	padding: rem( 10px ) rem( 24px ) rem( 24px );
+	padding: rem( 10px ) rem( 30px ) rem( 25px );
```
The `max-width` in combination with the `padding` creates a fairly narrow space for the text on small devices. WDYT about removing the `max-width` and reducing the horizontal `padding`? If we are concerned about illegible line length, we could apply an `em`-based `max-width` to `.jp-apps-card__header`/`.jp-apps-card__paragraph` instead. 

<img width="322" alt="Narrow small screen content width" src="https://user-images.githubusercontent.com/438664/192859059-a7d491a4-758a-4083-ac57-c85bbb649f5c.png">

SiobhyB replied:
@dcalhoun, thank you! I implemented those suggested changes in https://github.com/Automattic/jetpack/pull/26276/commits/52bcdf75ac819c1fd258e0cf2c449eed82748d18. I didn't see a need to set a max-width to `max-width` to `.jp-apps-card__header`/`.jp-apps-card__paragraph` as, even with removing the `max-width` from the container and reducing the horizontal padding, the padding seems enough to prevent illegible text.

Tested with a Galaxy Fold emulator _(280 x 683)_:

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/193103345-126cd768-8c51-4f7e-aad4-c41127107a22.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/193102590-65176fe5-17a6-4c62-b93c-f602bc606098.png" width="100%"> |

Let me know if you think this needs further tweaking!

dcalhoun replied:
Looks good! 🙇🏻

dcalhoun reviewed (dismissed):
🚀 LGTM. I tested in macOS Safari for various viewport widths. I left a couple of notes that I do not consider blockers. 

Also, we may be able to reduce the size of the SVG asset with a tool like [SVGOMG](https://jakearchibald.github.io/svgomg/).

SiobhyB committed:
Merge branch 'trunk' into rnmobile/update-get-apps-section

SiobhyB committed:
Optimize URL for SEO

The update URL in this commit is the one recommended for use by the Jetpack team.

Co-authored-by: David Calhoun <438664+dcalhoun@users.noreply.github.com>

SiobhyB committed:
Add clarification to instructions

Co-authored-by: David Calhoun <438664+dcalhoun@users.noreply.github.com>

SiobhyB committed:
Optimize SVGs for performance

With this commit, I made use of an online tool to reduce the size of the SVGs.

SiobhyB committed:
Implement style tweaks to avoid super narrow views

As per the following feedback, the app card appeared super narrow on smaller devices: https://github.com/Automattic/jetpack/pull/26276#discussion_r982730130

This commit introduces some smaller tweaks to prevent that.

SiobhyB committed:
Prevent splitting strings for clearer translations

The 'createInterpolateElement' function enables an inline link to be added to a string without splitting that string up into various HTML elements, which will help translators to more easily interpret the full meaning of the string.

SiobhyB committed:
Merge branch 'trunk' into rnmobile/update-get-apps-section

SiobhyB commented:
Thank you again for your review @dcalhoun, I believe I've addressed all your points now, including optimizing the SVGs (done in https://github.com/Automattic/jetpack/pull/26276/commits/e80f5e60db205f6859391226e40faaed438df742). I've also made a change in https://github.com/Automattic/jetpack/pull/26276/commits/6d5ccf45f41e44a286f1041d0109e74250a23f25, which addresses an issue with the way I was handling the strings. If the latest version looks good, I'll aim to request a final review from the Jetpack crew tomorrow. 🙇‍♀️

bindlegirl commented:
@SiobhyB This is looking good and thanks for the screenshots!

Recently we merged a PR that introduces device detection functionality that helps display a correct App badge on mobile web. It's used on the Recommendations tab. Do you think it could be useful in this case as well?

PR: https://github.com/Automattic/jetpack/pull/26093

SiobhyB committed:
Display app-specific badges

With this commit, the 'detectMobileDevice' function is used to display either the App Store or Google Play badges, depending on whether a user is viewing the badge on a iOS or Android device.

SiobhyB committed:
Remove uneccessary top margin from apps badge

Now that only one badge will display (depending on the platform the user's viewing it on), there is no need to add margin to space out two badge.

SiobhyB commented:
@bindlegirl, love that suggestion! I've gone ahead to implement it in https://github.com/Automattic/jetpack/pull/26276/commits/6e17e6f06344327f7c75fa5e06da997365d2cd34. Let me know how that seems to you:

| Pixel 5  | iPhone XR |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/193277291-d2b59e55-6fb7-414a-991a-cd9cdc855ec2.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/193277046-9fae6083-fd0b-4ef3-a3be-e63560e4aa93.png" width="100%"> |

bindlegirl reviewed (approved):
Looks good!

samiff commented:
cherry-picked to `jetpack/branch-11.4 ` in 3dbedf0e47b24e95247f137de90f525da4d76a98

fluiddot commented on projects/plugins/jetpack/_inc/client/components/apps-card/index.jsx:
```diff
@@ -24,17 +24,62 @@ class AppsCard extends React.Component {
+						'jetpack'
+					),
+					{
+						a: <a className="jp-apps-card__link" href="https://jetpack.com/app" />,
```
@SiobhyB  Looks like we should use the [Jetpack Redirects helper](https://github.com/Automattic/jetpack/tree/trunk/projects/js-packages/components/tools/jp-redirect) when adding external links instead of direct linking.

SiobhyB replied:
@fluiddot, thanks! Addressed in https://github.com/Automattic/jetpack/pull/26668.
